src/dnx/pal.win32.cpp

```diff
@@ -81,19 +83,65 @@ BOOL GetFullPath(LPCTSTR szPath, LPTSTR pszNormalizedPath)
     return TRUE;
 }
 
+namespace
+{
+    std::wstring get_runtime_path(TraceWriter& trace_writer)
+    {
+        std::vector<wchar_t*> servicing_locations =
+        {
+            L"DNX_SERVICING",
+            L"ProgramFiles(x86)"
+        };
+
+        wchar_t servicing_location_buffer[MAX_PATH];
+        // The servicing index should be always under %ProgramFiles(x86)% however
+        // on 32-bit OSes there is only %ProgramFiles%
+        if (GetEnvironmentVariable(L"ProgramFiles(x86)", servicing_location_buffer, MAX_PATH) == 0)
```

Comment:
I did not do it since it is a bit pain for native code. They shouldn't be `#define`s but `const`s once you make them consts they should not be in .h (one definition rule) so you have to create a *.cpp file and then use `extern` forward declaration to be able to use it. Because we are using these names only in one place and all but one is used just once I did not bother to do that. If you feel strongly about that I can for now define the consts in this file and when it turns out we need to use them elsewhere the can be moved to a separate file.